### ISSUE TITLE
Error: unknown command "runstate" for "git-town config"

### ISSUE THREAD

**Issue by JonasOstir** [git-town/git-town#2282]
When trying to ship a nested feature I got this error message: 
```
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

Error: exit status 1

To abort, run "git-town abort".
To continue after having resolved conflicts, run "git-town continue".
```

I was back on my main branch

After:
```
git town continue

Error: cannot load previous run state: cannot parse content of file "/Users/[user]/Library/Application Support/git-town/runstate/users-[user]-[directory].json": unknown step type: "*UpdateProposalTargetStep", run "git town config runstate reset" to reset it
```
[user] => user directory
[directory] => path-to-json

Trying to reset:
```
git town config runstate reset

Error: unknown command "runstate" for "git-town config"
```

After removing the step with the *UpdateProposalTargetStep type I can do a `git town abort`

**Comment by kevgo** [comment]
Interesting edge case, thanks for reporting it. Can you please provide more details about your particular setup?

- Which branch did you ship, what were its parent and child branches?
- What are possible reasons why merging updates from the remote branch didn't work on your feature branch? (see https://github.com/git-town/git-town/blob/2cd0d3611826dcfb49ea83d29654bd68a1169899/features/ship/current_branch/ship_current_branch.feature#L17)? 

I have created #2288 to address some of the other errors you encountered and #2290 to add the missing "git-town config runstate reset" command.

**Comment by JonasOstir** [comment]
Thank you for looking into it!

> Which branch did you ship, what were its parent and child branches?
It was something like:
```
main
  JIRA-1
    JIRA-2
      JIRA-3
        JIRA-4
```
Where I first tried to ship `JIRA-3` which wasn't working (should have read the documentation 😅 ). Then I tried shipping `JIRA-1`. In the end I could merge the whole feature tree with a sporadic removal of the `UpdateProposalTargetStep` step.

> What are possible reasons why merging updates from the remote branch didn't work on your feature branch? 

Good question. I try to use the `git town sync` command as much as possible to be up to date. It could be unrelated but we do a "Squash on merge". It could also happen that a branch was updated with main through the "Update branch" button but can't say for sure.

**Comment by JonasOstir** [comment]
Just had it again with a new feature branch:
```
main
  feature-1
    feature-2
```
did a `git town ship` on `feature-1` which was successful. Checked out `feature-2` and tried a `git town sync` which resulted in:
```
git town sync

Error: cannot load previous run state: cannot parse content of file "/Users/[user]/Library/Application Support/git-town/runstate/users-[user]-[directory].json": unknown step type: "*UpdateProposalTargetStep", run "git town config runstate reset" to reset it
```

**Comment by kevgo** [comment]
Apologies for the ongoing problems with the `UpdateProposalTargetStep`. I have released v9.0.1 that should fix this error. Please let me know if this issue happens still with the new version.

There is another still unfixed bug lurking in there that happens in an edge case that you encounter. I'd still like to reproduce and fix it. So please, the next time you encounter this, please post the full output of Git Town (with sensitive data removed). Thanks!

**Comment by kevgo** [comment]
`git sync` received a major overhaul in the upcoming v10 release. This should be fixed now.

**Comment by kevgo** [comment]
FYI the fix for this has been released as [v10.0.0](https://github.com/git-town/git-town/releases/tag/v10.0.0). Apologies for the delay, this PR came in the middle of a large-scale product cleanup that involved a number of breaking changes that need to be shipped together but took a while to finalize and test.
